llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -22137,6 +22137,22 @@ static SDValue performExtendCombine(SDNode *N,
       N->getOperand(0)->getOpcode() == ISD::SETCC)
     return performSignExtendSetCCCombine(N, DCI, DAG);
 
+  // If we see (any_extend (bswap ...)) with bswap returning an i16, we know
+  // that the top half of the result register must be unused, due to the
+  // any_extend. This means that we can replace this pattern with (rev16
+  // (any_extend ...)). This saves a machine instruction compared to (lsr (rev
+  // ...)), which is what this pattern would otherwise be lowered to.
+  if (N->getOpcode() == ISD::ANY_EXTEND &&
+      N->getOperand(0).getOpcode() == ISD::BSWAP &&
+      N->getOperand(0).getValueType().isScalarInteger() &&
+      N->getOperand(0).getValueType().getFixedSizeInBits() == 16) {
+    SDNode *BswapNode = N->getOperand(0).getNode();
```

Comment:
Usually we would do `SDLoc DL(N);`, and use that elsewhere in both the nodes.

Reply:
By this do you mean just replacing the inline `SDLoc(N)` in the second `DAG.getNode` with `DL`, defined as you suggest?
I'm not sure how else I could use `SDLoc DL(N)` - The first `DAG.getNode` uses `SDLoc(BswapNode)`, not `SDLoc(N)`, and I also use the definition of `BswapNode` in the `BswapNode->getOperand(0)` expression, not just as a location.

Reply:
Change `SDLoc(BswapNode)` with `DL` in the getNode calls and define:

    SDLoc DL(N);